core/src/main/java/org/apache/carbondata/core/util/ByteUtil.java

```diff
@@ -465,6 +465,39 @@ public static short toShort(byte[] bytes, int offset, final int length) {
   }
 
   /**
+   * int => byte[3]
+   * supported range is [-8388607, 8388607], note that Math.pow(2, 24) == 8388608
+   */
+  public static byte[] to3Bytes(int val) {
+    assert val <= Math.pow(2, 23) && val >= (-Math.pow(2, 23) - 1);
+
+    int value = val < 0 ? -val : val;
+    byte[] b = new byte[3];
+    b[0] = (byte) (value & 0xFF);
+    b[1] = (byte) ((value >>> 8) & 0xFF);
+    b[2] = (byte) ((value >>> 16) & 0x7F);
```

Comment:
Because in line474, value is converted to absolute value, so in line479 it needs to keep the "sign bit", otherwise the "sign bit" information will be lost